### DISCUSSION TITLE
validate_play_store_json_key issues

### DISCUSSION THREAD

**Discussion by lawrence-c** [fastlane/fastlane#18108] [Q&A] [answered]
Hi,

I'm new with using `fastlane`, and have hit a small issue with validating the play store key. I have installed `fastlane` via `brew` on macOS.

Following the instructions from https://docs.fastlane.tools/getting-started/android/setup/#setting-up-supply, I manage to do everything up to step 7, and then I don't have any option for choosing a Role, instead I see a `Permissions` tab with tabs for specific `App permissions` and `Account permissions`. I've ticked nearly all permissions and have tried to continue.

When running `fastlane run validate_play_store_json_key json_key:/path/to/redacted/service_account_key.json --verbose`, I get the following error:

```
+----------+--------------------------------------------------+
|          Summary for validate_play_store_json_key           |
+----------+--------------------------------------------------+
| json_key | /path/to/redacted/service_account_key.json.      |
| timeout  | 300                                              |
+----------+--------------------------------------------------+

DEBUG [2021-02-03 16:54:43.87]: Fetching a new access token from Google...
ERROR [2021-02-03 17:00:48.42]: Could not establish a connection to Google Play Store with this json key file.
ERROR [2021-02-03 17:00:48.43]: Unexpected error: #<Faraday::ConnectionFailed wrapped=#<Net::OpenTimeout: execution expired>>
/usr/local/Cellar/fastlane/2.172.0/libexec/gems/googleauth-0.14.0/lib/googleauth/signet.rb:118:in `rescue in retry_with_error'
/usr/local/Cellar/fastlane/2.172.0/libexec/gems/googleauth-0.14.0/lib/googleauth/signet.rb:107:in `retry_with_error'
/usr/local/Cellar/fastlane/2.172.0/libexec/gems/googleauth-0.14.0/lib/googleauth/signet.rb:80:in `fetch_access_token!'
/usr/local/Cellar/fastlane/2.172.0/libexec/gems/fastlane-2.172.0/supply/lib/supply/client.rb:53:in `initialize'
/usr/local/Cellar/fastlane/2.172.0/libexec/gems/fastlane-2.172.0/supply/lib/supply/client.rb:19:in `new'
/usr/local/Cellar/fastlane/2.172.0/libexec/gems/fastlane-2.172.0/supply/lib/supply/client.rb:19:in `make_from_config'
/usr/local/Cellar/fastlane/2.172.0/libexec/gems/fastlane-2.172.0/fastlane/lib/fastlane/actions/validate_play_store_json_key.rb:14:in `run'
/usr/local/Cellar/fastlane/2.172.0/libexec/gems/fastlane-2.172.0/fastlane/lib/fastlane/runner.rb:263:in `block (2 levels) in execute_action'
/usr/local/Cellar/fastlane/2.172.0/libexec/gems/fastlane-2.172.0/fastlane/lib/fastlane/actions/actions_helper.rb:69:in `execute_action'
/usr/local/Cellar/fastlane/2.172.0/libexec/gems/fastlane-2.172.0/fastlane/lib/fastlane/runner.rb:255:in `block in execute_action'
/usr/local/Cellar/fastlane/2.172.0/libexec/gems/fastlane-2.172.0/fastlane/lib/fastlane/runner.rb:229:in `chdir'
/usr/local/Cellar/fastlane/2.172.0/libexec/gems/fastlane-2.172.0/fastlane/lib/fastlane/runner.rb:229:in `execute_action'
/usr/local/Cellar/fastlane/2.172.0/libexec/gems/fastlane-2.172.0/fastlane/lib/fastlane/one_off.rb:42:in `run'
/usr/local/Cellar/fastlane/2.172.0/libexec/gems/fastlane-2.172.0/fastlane/lib/fastlane/one_off.rb:22:in `execute'
/usr/local/Cellar/fastlane/2.172.0/libexec/gems/fastlane-2.172.0/fastlane/lib/fastlane/commands_generator.rb:224:in `block (2 levels) in run'
/usr/local/Cellar/fastlane/2.172.0/libexec/gems/commander-fastlane-4.4.6/lib/commander/command.rb:178:in `call'
/usr/local/Cellar/fastlane/2.172.0/libexec/gems/commander-fastlane-4.4.6/lib/commander/command.rb:153:in `run'
/usr/local/Cellar/fastlane/2.172.0/libexec/gems/commander-fastlane-4.4.6/lib/commander/runner.rb:476:in `run_active_command'
/usr/local/Cellar/fastlane/2.172.0/libexec/gems/fastlane-2.172.0/fastlane_core/lib/fastlane_core/ui/fastlane_runner.rb:76:in `run!'
/usr/local/Cellar/fastlane/2.172.0/libexec/gems/commander-fastlane-4.4.6/lib/commander/delegates.rb:15:in `run!'
/usr/local/Cellar/fastlane/2.172.0/libexec/gems/fastlane-2.172.0/fastlane/lib/fastlane/commands_generator.rb:352:in `run'
/usr/local/Cellar/fastlane/2.172.0/libexec/gems/fastlane-2.172.0/fastlane/lib/fastlane/commands_generator.rb:41:in `start'
/usr/local/Cellar/fastlane/2.172.0/libexec/gems/fastlane-2.172.0/fastlane/lib/fastlane/cli_tools_distributor.rb:122:in `take_off'
/usr/local/Cellar/fastlane/2.172.0/libexec/gems/fastlane-2.172.0/bin/fastlane:23:in `<top (required)>'
/usr/local/Cellar/fastlane/2.172.0/libexec/bin/fastlane:23:in `load'
/usr/local/Cellar/fastlane/2.172.0/libexec/bin/fastlane:23:in `<main>'
INFO [2021-02-03 17:00:48.43]: Result: true
```

Any ideas what I could have misconfigured here?

Thanks in advance!

---

### Stack

| Key                         | Value                                                   |
| --------------------------- | ------------------------------------------------------- |
| OS                          | 10.15.6                                                 |
| Ruby                        | 2.7.2                                                   |
| Bundler?                    | false                                                   |
| Git                         | git version 2.30.0                                      |
| Installation Source         | /usr/local/Cellar/fastlane/2.172.0/libexec/bin/fastlane |
| Host                        | Mac OS X 10.15.6 (19G2021)                              |
| Ruby Lib Dir                | /usr/local/Cellar/ruby@2.7/2.7.2/lib                    |
| OpenSSL Version             | OpenSSL 1.1.1i  8 Dec 2020                              |
| Is contained                | false                                                   |
| Is homebrew                 | true                                                    |
| Is installed via Fabric.app | false                                                   |
| Xcode Path                  | /Library/Developer/CommandLineTools/                    |
| Xcode Version               |                                                         |

**Comment by lawrence-c** [comment, marked as answer]
Looks like the issue has something to do with being on a VPN. When I'm not a VPN, this validates fine... 🤔

**Comment by sreeraj-ramesh** [comment]
Im getting error like this while using the command 

**fastlane run validate_play_store_json_key json_key:/usr/local/V1/pc-api-6240919-256-3ef990fe8941.json --verbose**

DEBUG [2021-12-10 10:39:12.58]: Fetching a new access token from Google...
ERROR [2021-12-10 10:39:17.55]: Could not establish a connection to Google Play Store with this json key file.
ERROR [2021-12-10 10:39:17.55]: Unexpected error: #<Faraday::ConnectionFailed wrapped=#<Errno::ECONNRESET: Connection reset by peer - SSL_connect>>
/var/lib/gems/2.5.0/gems/googleauth-1.1.0/lib/googleauth/signet.rb:112:in `rescue in retry_with_error'
/var/lib/gems/2.5.0/gems/googleauth-1.1.0/lib/googleauth/signet.rb:101:in `retry_with_error'
/var/lib/gems/2.5.0/gems/googleauth-1.1.0/lib/googleauth/signet.rb:74:in `fetch_access_token!'
/var/lib/gems/2.5.0/gems/fastlane-2.198.1/supply/lib/supply/client.rb:53:in `initialize'
/var/lib/gems/2.5.0/gems/fastlane-2.198.1/supply/lib/supply/client.rb:19:in `new'
/var/lib/gems/2.5.0/gems/fastlane-2.198.1/supply/lib/supply/client.rb:19:in `make_from_config'
/var/lib/gems/2.5.0/gems/fastlane-2.198.1/fastlane/lib/fastlane/actions/validate_play_store_json_key.rb:14:in `run'
/var/lib/gems/2.5.0/gems/fastlane-2.198.1/fastlane/lib/fastlane/runner.rb:263:in `block (2 levels) in execute_action'
/var/lib/gems/2.5.0/gems/fastlane-2.198.1/fastlane/lib/fastlane/actions/actions_helper.rb:69:in `execute_action'
/var/lib/gems/2.5.0/gems/fastlane-2.198.1/fastlane/lib/fastlane/runner.rb:255:in `block in execute_action'
/var/lib/gems/2.5.0/gems/fastlane-2.198.1/fastlane/lib/fastlane/runner.rb:229:in `chdir'
/var/lib/gems/2.5.0/gems/fastlane-2.198.1/fastlane/lib/fastlane/runner.rb:229:in `execute_action'
/var/lib/gems/2.5.0/gems/fastlane-2.198.1/fastlane/lib/fastlane/one_off.rb:42:in `run'
/var/lib/gems/2.5.0/gems/fastlane-2.198.1/fastlane/lib/fastlane/one_off.rb:22:in `execute'
/var/lib/gems/2.5.0/gems/fastlane-2.198.1/fastlane/lib/fastlane/commands_generator.rb:225:in `block (2 levels) in run'
/var/lib/gems/2.5.0/gems/commander-4.6.0/lib/commander/command.rb:187:in `call'
/var/lib/gems/2.5.0/gems/commander-4.6.0/lib/commander/command.rb:157:in `run'
/var/lib/gems/2.5.0/gems/commander-4.6.0/lib/commander/runner.rb:444:in `run_active_command'
/var/lib/gems/2.5.0/gems/fastlane-2.198.1/fastlane_core/lib/fastlane_core/ui/fastlane_runner.rb:117:in `run!'
/var/lib/gems/2.5.0/gems/commander-4.6.0/lib/commander/delegates.rb:18:in `run!'
/var/lib/gems/2.5.0/gems/fastlane-2.198.1/fastlane/lib/fastlane/commands_generator.rb:353:in `run'
/var/lib/gems/2.5.0/gems/fastlane-2.198.1/fastlane/lib/fastlane/commands_generator.rb:42:in `start'
/var/lib/gems/2.5.0/gems/fastlane-2.198.1/fastlane/lib/fastlane/cli_tools_distributor.rb:122:in `take_off'
/var/lib/gems/2.5.0/gems/fastlane-2.198.1/bin/fastlane:23:in `<top (required)>'
/usr/local/bin/fastlane:23:in `load'
/usr/local/bin/fastlane:23:in `<main>'
INFO [2021-12-10 10:39:17.55]: Result: true
DEBUG [2021-12-10 10:39:17.55]: All plugins are up to date
